native/native_main.nim

```diff
@@ -0,0 +1,97 @@
+# compile with 
+# nim c --verbosity:0 native_main.nim
+
+# test with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ./native_main
+
+# compare with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | python ~/.local/share/tridactyl/native_main.py
+
+import json
+import options
+import osproc
+# import endians
+import struct # nimble install struct
+
+let VERSION = "0.2.0"
+
+type 
+    MessageRecv* = object
+        cmd*, version*, content*, error*, command*: Option[string]
+        code: Option[int]
+
+# let a = MessageResp(cmd: some(""))
+# echo(a)
+
+proc getMessage(): MessageRecv =
+
+    # length of the string - not required AFAICT
+    var length: int32
+    discard readBuffer(stdin, addr(length), 4)
+
+    return to(parseJson(readAll(stdin)),MessageRecv)
+
+proc handleMessage(msg: MessageRecv): string =
+
+    let cmd = msg.cmd.get()
+    var command: string
+
+    case cmd:
+        of "run":
+            command = "{\"cmd\": \"run\", \"content\": \"" & $ (execProcess(msg.command.get(), options={poEvalCommand,poStdErrToStdOut})) & "\", \"code\": 0}"
+
+        of "version":
+            command = "{\"version\": \"" & VERSION & "\"}"
+
+    return command
+
+let message = handleMessage(getMessage())
+
+# this works fine V so I reckon the problem must be with the "nulls" it spits out
+# could just write our own JSON? ... is that so mad?
+# let message = "{\"version\": \"0.1.11\"}" #$ %* handleMessage(getMessage()) # %* converts the object to JSON
+
+# but this doesn't work -_-
+# bleugh
+# let message = "{\"version\": \"0.2.0\"}" #$ %* handleMessage(getMessage()) # %* converts the object to JSON
```

Comment:
I don't like computers

Reply:
https://www.youtube.com/watch?v=jpoFKZlQPV4

Reply:
Fixed it - I waited 15 hours and tried again and now it works. hooray